Extension/src/LanguageServer/client.ts

```diff
@@ -1799,9 +1799,38 @@ export class DefaultClient implements Client {
                             const candidate: string = response.candidates[i];
                             const tuUri: vscode.Uri = vscode.Uri.parse(candidate);
                             if (await provider.canProvideConfiguration(tuUri, tokenSource.token)) {
-                                const configs: SourceFileConfigurationItem[] = await provider.provideConfigurations([tuUri], tokenSource.token);
+                                const configs: util.Mutable<SourceFileConfigurationItem>[] = await provider.provideConfigurations([tuUri], tokenSource.token);
                                 if (configs && configs.length > 0 && configs[0]) {
-                                    return configs;
+                                    const fileConfiguration: configs.Configuration | undefined = this.configuration.CurrentConfiguration;
+                                    if (fileConfiguration?.mergeConfigurations) {
+                                        configs.forEach(config => {
+                                            for (const p in fileConfiguration.includePath) {
+                                                if (!config.configuration.includePath.includes(p)) {
+                                                    config.configuration.includePath.push(p);
+                                                }
+                                            }
+
+                                            for (const d in fileConfiguration.defines) {
```

Comment:
@willson556 
Thanks for the addressing the change requests.

I tested the latest changes and found the `d` variable in `for (const d in fileConfiguration.defines)` was returning the index number of array `fileConfiguration.defines` instead of the values in the array. It was also the same case for `for (const p in fileConfiguration.includePath)` and probably for `for (const i in fileConfiguration.forcedInclude)`.  Could you check if this is also happening when you test and debug it? Unless I'm wrong, the Typescript syntax seems to be correct for accessing the values in an array.

Reply:
I got bit by this in CMake Tools once.  We added the linter rule `"@typescript-eslint/no-for-in-array": "error",`
cpptools should add this rule too.

Reply:
Sorry, should have retested -- that looked like a straightforward change to satisfy the linter. I've put it back to a `.foreach()` call but with explicit `if (array) {}` checks to avoid confusing the linter.

Reply:
I retested the changes. It works now. Thank you!